src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -101,5 +103,76 @@ public void TranslateTriplePatternTest() {
 		
 		return new Pair<>(testSet, expectedRes);
 	}
+	*/
+	
+	@Test
+	public void TranslateSolutionMappingTest() {
+		final Set<Triple> testSet = new HashSet<>();
+		Node s = NodeFactory.createURI("a");
+		Node p = OWL.equivalentClass.asNode();
+		Node o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("b");
+		p = OWL.unionOf.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("c");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = RDF.nil.asNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("c");
+		p = OWL.equivalentClass.asNode();
+		o = NodeFactory.createURI("m");
+		testSet.add(new Triple(s, p ,o));
+		
+		final VocabularyMapping vm = new VocabularyMappingImpl(testSet);
+		
+		final BindingBuilder testBuilder = BindingBuilder.create();
+		Var testVar = Var.alloc("v");
+		Node testNode = NodeFactory.createURI("n");
+		testBuilder.add(testVar, testNode);
+		testVar = Var.alloc("w");
+		testNode = NodeFactory.createURI("m");
+		testBuilder.add(testVar, testNode);
+		
+		final SolutionMapping testSm = new SolutionMappingImpl(testBuilder.build());		
+		Set<SolutionMapping> translation = vm.translateSolutionMapping(testSm);
+		
+		Set<SolutionMapping> expectedResults = new HashSet<>();
+		final BindingBuilder first = BindingBuilder.create();
+		Node n = NodeFactory.createURI("a");
+		first.add(Var.alloc("v"), n);
+		n = NodeFactory.createURI("c");
+		first.add(Var.alloc("w"), n);
+		expectedResults.add(new SolutionMappingImpl(first.build()));
+		
+		final BindingBuilder second = BindingBuilder.create();
+		n = NodeFactory.createURI("b");
+		second.add(Var.alloc("v"), n);
+		n = NodeFactory.createURI("c");
+		second.add(Var.alloc("w"), n);
+		expectedResults.add(new SolutionMappingImpl(second.build()));
+		
+		System.out.print(expectedResults.toString());
+		System.out.print(translation.toString());
+		
+		//assertEquals(expectedResults, translation);
```

Comment:
Do you know the cause this doesn't work?

Reply:
When trying I, had this issue with the SolutionMapping equals function not being called (no prints or changes in i mattered). I tried comparing sizes and doing containsAll but that didn't work either.

Reply:
I will check what is going on there. I will leave a new comment on the meantime.